pkg/cmd/kind/delete/cluster/deletecluster.go

```diff
@@ -61,6 +66,19 @@ func NewCommand(logger log.Logger, streams cmd.IOStreams) *cobra.Command {
 		"",
 		"sets kubeconfig path instead of $KUBECONFIG or $HOME/.kube/config",
 	)
+
+	cobra.CheckErr(cmd.RegisterFlagCompletionFunc(name, func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		clusters, err := cluster.NewProvider(
+			cluster.ProviderWithLogger(logger),
```

Comment:
What happens when podman support is stable and we move from `KIND_EXPERIMENTAL_PROVIDER` to `KIND_PROVIDER` as a default for a `--provider` flag? Can flag completion check for other flag values?

This feature seems like it may unfortunately be at least as problematic to maintain as it is useful.

Using `KIND_CLUSTER_NAME` is another option to get around repeatedly typing these.

Reply:
> Can flag completion check for other flag values?

Yes it can.

I think the best solution is to extract ·get clusters· as a common function. This way we can only maintain the code in the same place

Reply:
> Does this mean that the implementation of kind get clusters will change? The code in PR is copied from get clusters

Yes. It's defaulting based on availability or the env currently.

Reply:
And how about `extract kind get clusters runE as a common function` ? IMO, the completion is like an alias of `get clusters`.

Reply:
Sure, I've just never used a tool where one flag completed based on another flag and didn't know if cobra can support this. If it can then let's go ahead :-)